File: 0938-range-sum-of-bst/0938-range-sum-of-bst.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    var sum = 0
    fun rangeSumBST(root: TreeNode?, low: Int, high: Int): Int {
        sum = 0
        dfs(root, low, high)
        return sum
    }
    fun dfs(root: TreeNode?, low: Int, high: Int) {
        if (root == null) return
        if (root.`val` >= low && root.`val` <= high) sum+=root.`val`
        dfs(root.left, low, high)
        dfs(root.right, low, high)
    }
}

File: 0938-range-sum-of-bst/0938-range-sum-of-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        res = 0
        if not root:
            return 0
        
        if root.val >= low and root.val <= high:
            res += root.val
        if root.val >= low:
            res += self.rangeSumBST(root.left, low, high)
        if root.val <= high:
            res += self.rangeSumBST(root.right, low, high)

        return res